inst/include/Rcpp/Environment.h

```diff
@@ -246,9 +246,8 @@ namespace Rcpp{
                        we have to go back to R to do this operation */
                     SEXP internalSym = Rf_install( ".Internal" );
                     SEXP removeSym = Rf_install( "remove" );
-                    Shield<SEXP> call( Rf_lang2(internalSym,
-                            Rf_lang4(removeSym, Rf_mkString(name.c_str()), Storage::get__(), Rf_ScalarLogical( FALSE ))
-                        ) );
+                    Shield<SEXP> str(Rf_mkString(name.c_str()));
+                    Shield<SEXP> call(Rf_lang2(internalSym, Rf_lang4(removeSym, str, Storage::get__(), Rf_ScalarLogical(FALSE))));
```

Comment:
It may also be worth pulling out `Rf_lang4()` from this call and protecting that as well, since IIUC that will be an unprotected R object being passed into `Rf_lang2()`, and so _in theory_ could be cleaned up by the GC.

Reply:
Hmm, I might be wrong. IIUC, each of the `lang*` helper functions protects the arguments passed in, before the final call is created:

https://github.com/wch/r-source/blob/eafae2b2ed39b3383943cd6c36f92d78f966811e/src/include/Rinlinedfuns.h#L629-L677

So this pattern should in fact be safe.